wrappers/services.go

```diff
@@ -191,6 +191,7 @@ Requires=snapd.frameworks.target{{ if .App.Socket }} {{.SocketFileName}}{{end}}
 X-Snappy=yes
 
 [Service]
+ExecPreStart=mkdir -p {{.App.Snap.RootUserDataDir}} {{.App.Snap.RootCommonUserDataDir}}
```

Comment:
@niemeyer Indeed, this is just a temporary step, in `snap run` we have the ability to create the user data dirs (cf. https://github.com/snapcore/snapd/blob/master/cmd/snap/cmd_run.go#L125). For `snap run` we need a new stable ubuntu-core snap that contains snapd 2.11. This is really close, we need to create a candidate core snap with snapd 2.11 and then do a QA run on this. This is the first time we do a stable core update so we need to be a bit careful. Once that is in we can land `mvo5:snapd/snap-run3` and this all goes away and `snap run` takes care of it.